### PR TITLE
support logging of orb-slam2 statistic

### PR DESCRIPTION
**Pull request by apattnaik0721013**
- #76 is addressed.
- boost::log feature used to provide logging in orb-slam2.
- orb-slam2 entities (frame,local keyframes ,keyframes, local map-points, map-point) are displayed in the log.

### CODE REVIEW

**Comment by apattnaik0721013** [comment]
sample log( [orbslam2_garching_3.log](https://github.com/paul-michalik/ORB_SLAM2/files/1980264/orbslam2_garching_3.log) ) is created from garching-3 images.

File: CMakeLists.txt [hunk 1]
```diff
@@ -53,7 +53,9 @@ add_library(orb-slam2-ext STATIC
     ext/app_monitor_api.h
     ext/app_monitor_api_impl.h
     ext/messages.h
-    ext/functionless-sentinel.cpp)
+    ext/functionless-sentinel.cpp
+    ext/statistics.h	
+    ext/statistics.cc)
```
Comment (paul-michalik) [inline CMakeLists.txt]: Please use unified extensions for ***.cpp** source files!

File: ext/statistics.h [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+			>::type
+			>
+		{
+		private:
+			// The file to write the collected information to
+			std::ofstream m_csv_file;
+		public:
+			// The constructor initializes the internal data
+			explicit stat_collector(std::string &file_name):m_csv_file(file_name, std::ios::out)
+			{ if (!m_csv_file.is_open()) throw std::runtime_error("could not open the CSV file"); }
```
Comment (paul-michalik) [inline ext/statistics.h]: Please write every statement on separate line!

File: ext/statistics.h [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+#include <boost/log/sinks/basic_sink_backend.hpp>
+#include <boost/log/sources/logger.hpp>
+#include <boost/log/sources/record_ostream.hpp>
+#include <iostream>
+#include <string>
+
+namespace ORB_SLAM2
+{
+	namespace ext {
+
```
Comment (paul-michalik) [inline ext/statistics.h]: Please do not alias namespaces like that. I'd prefer no aliasing at all, but if you insist use `namespace log = boost::log;` and be explicit about the others, i.e. `log::sources`

File: src/System.cc [hunk 4]
```diff
@@ -426,6 +424,18 @@ void System::SaveKeyFrameTrajectoryTUM(const string &filename)
...
+    for (auto p_map_point : vpMP)
+    {
+        if (p_map_point)
+        {
+            if (p_map_point->is_semantic())
+                semantic_mappoints++;
+        }
+    }
+
+    ext::statistics::get().update_status(vpKFs.size(), vpMP.size(), semantic_mappoints);
```
Comment (paul-michalik) [inline src/System.cc]: This isn't very concise... If you want to provide a semantic logger API which understands the domain, please factor the entire logic of fetching the map points and dump them into a function, i.e. something like:
```
ext::statistics::get().log_map_state(mpMap);
```

File: ext/statistics.h [hunk 5]
```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#include <atomic>
+#include <vector>
+#include <opencv2/core/types.hpp>
+
+#include "Map.h"
```
Comment (paul-michalik) [inline ext/statistics.h]: If you decide to pull in everything from `boost/log` here, why is it necessary to split between header and implementation? Either hide implementation details in an implementation file and expose what is the public logger API here, or leave everything in one file...

**Comment by apattnaik0721013** [comment]
@paul-michalik 

- i have addressed all the review comments.
- sample generated stats file [orbslam.xlsx](https://github.com/paul-michalik/ORB_SLAM2/files/2016864/orbslam.xlsx) is attached.
Note: 
- as i am not able upload csv file hence formatted xlsx file is attached.
- this stats is generated from Garching loop closer 3 images with semantic data.

**Comment by paul-michalik** [comment]
> as i am not able upload csv file hence formatted xlsx file is attached.

What do you mean you are unable to upload csv file...? :-)

**Pull request merged**